workshop/serverless/01-serverless.md

```diff
@@ -0,0 +1,49 @@
+# Install and configure Serverless
+
+
+**What is Serverless?** \
+Who could be better explaining this than Serverless:\
+*"Just like wireless internet has wires somewhere, serverless architectures still have servers somewhere. What ‘serverless’ really means is that, as a developer you don’t have to think about those servers. You just focus on code.
+This was initially embodied in function-as-a-service infrastructure, like AWS Lambda, that had Serverless properties like: function as the unit of deployment, 100% auto-scaling, and pay-per-execution pricing models. More recently many other types of ‘serverless infrastructure’ have emerged such as zero configuration databases, APIs for handling user management, and highly abstracted container orchestration engines. At the end of the day, Serverless is any developer tooling which the developer does not need to worry at all about the underlying infrastructure."* \
+You can read more about this [here](https://serverless.com).
+
+
+## Install Serverless
+During the installation process you will be asked for your `AWS Access Key Id` and `AWS Secret Access Key` you got when you created the programmatic user. Take it at hand.
+
+1. Follow the insstallation instructions only [here](https://serverless.com/framework/docs/providers/aws/guide/quick-start/).
+2. If you didn't clone this repo yet, do it. If you alredy done go to `./infraestructure/aws/lambda` and take a look at the serverless yaml file [here](/infrastructure/aws/lambda/serverless.yml).
+There you could see the definition of your serverless service. Under the `functions` section you have the lambda function handler, this is the entry point for execution of your microservice. It also defines an HTTP method and path to trigger the lambda function.
+At the top of the file, there are some general parameters and there is also two IAM policies to allow access to parameters and encription keys needed by the lambda function to configure the database access.
+3. Take a look to the lambda function source [lambda_function.py](/infrastructure/aws/lambda/lambda_function.py). Check or update your paremeter's name if it is necessary.
+
+Now you are ready to deploy your serverless. Let's go!
+
+1. Under the directory `./infrastructure/aws/lambda/` \
+run: \
+`serverless deploy -v` \
+This will start to make a detailed set of instructions and will automatically create all that you need to have a working lambda function in your AWS account. At the end, if all works as expected, you will have the URL to access to your serveice.
+2. Now you can test your endpoint sending a GET request to you API endpoint and passing some string to the `query` request parameter. As a response you will get all the articles which has that string in their title. Now is time to go to the AWS console and take a look over the results of the serverless deployment.
```

Comment:
Explain from where they can get the endpoint (remember that you can find the entire endpoint of the API gateway on the AWS console). Explain that an API gateway was created and a link to API gateway documentation.